hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/table/action/commit/BaseSparkCommitActionExecutor.java

```diff
@@ -161,13 +166,14 @@ public HoodieWriteMetadata<HoodieData<WriteStatus>> execute(HoodieData<HoodieRec
     HoodieData<HoodieRecord<T>> inputRecordsWithClusteringUpdate = clusteringHandleUpdate(inputRecords);
 
     context.setJobStatus(this.getClass().getSimpleName(), "Building workload profile:" + config.getTableName());
-    HoodieTimer sourceReadAndIndexTimer = HoodieTimer.start(); // time taken from dedup -> tag location -> building workload profile
     WorkloadProfile workloadProfile =
```

Comment:
The new changes just initializes the timer on the fly and passed in, why not just move the timer at line 164 to the begining of the method?

Reply:
I moved it to the beginning of 
BaseWriteHelper.write

Reply:
addressed for all engines.

Reply:
Flink does not has the issue like lazy computation of Spark, so the timer does not make much sense, I think this is also true for Java, maybe we revert the changes for Flink and JAVA and just Spark as the minimum change